src/arrow_server.cpp

```diff
@@ -700,6 +701,15 @@ class UKVService : public arf::FlightServerBase {
             ukv_length_t* found_lengths = nullptr;
             ukv_octet_t* found_presences = nullptr;
             ukv_size_t tasks_count = static_cast<ukv_size_t>(input_batch_c.length);
+            strided_iterator_gt<ukv_key_t> input_keys(nullptr, sizeof(ukv_key_t));
+
+            /// @param `keys`
+            if (tasks_count) {
```

Comment:
Please bring back the keys parsing location and instead put an additional check on all `DoExchange` methods, that the table can't be empty.